src/lib/adapters/index.ts

```diff
@@ -0,0 +1,17 @@
+import type { ClientNoId } from '$lib/types'
+import SupabaseAdapter from './supabase'
+
+export interface Adapter {
+	// This is run when the app is mounted and should start app wide subscriptions
+	start?: () => Promise<void> | void
+	// This is run when the app unmounts and should clear subscriptions
+	stop?: () => Promise<void> | void
```

Comment:
I would just make these explicit and if the actual adapter has an empty implementation it's fine.

```suggestion
	start: () => Promise<void>
	// This is run when the app unmounts and should clear subscriptions
	stop: () => Promise<void>
```